slsim/lens.py

```diff
@@ -384,8 +384,21 @@ def point_source_magnitude(self, band, lensed=False, time=None):
                 )
                 return lensed_variable_magnitude
             else:
-                magnified_mag = self.source.point_source_magnitude(band) - magnif_log
-                return magnified_mag
+                if (
+                    isinstance(self.source.point_source_magnitude(band), np.ndarray)
```

Comment:
I am not sure this code needs to be there. Can't you make the Source() class such that the desired return statements are being given regardless of the variability model?

Reply:
it is also a very specific check and every other criteria is using the other convention. This should be the role of the Source class (or below that) to be handled and not the lens class

Reply:
This is required for supernovae because of its specific file format. It does not effect other cases. I moved this to source class.